### PR TITLE
RichCommand: Glyph + ContextMenu + EmptyRecycleBinAction

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
* Usage of glyphs in IAction. This simplifies the use of icons with the FontIcon and ColoredIcon properties of IRichCommand.
* Simplifies the creation of context menus. It is possible to pass a RichCommand to ContextMenuFlyoutItemViewModel to initialize it.
* Added RichCommand EmptyRecycleBin, which uses glyphs. This command is used everywhere, including the sidebar. Old commands are deleted. This new action does not yet use the IsExecutable property. This requires the context class and will be the subject of a future pr.
* A behavior has changed slightly. In the context menu of the RecycleBin page, the RecycleBin command is no longer active if the RecycleBin is empty. This seemed like a bug to me.
* A bug has been fixed in the parser of HotKey (not yet used).
* The next step is to add the context. This is the last part of the command base. Then, it will be necessary to convert the current commands (actions + use in the views).

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

**Comment by yaira2** [comment]
Can you add support for OpacityIcon? We're currently only using them in the toolbar but we're going to want to use them in the context menu as well.

File: src/Files.App/ViewModels/ContextMenuFlyoutItemViewModel.cs [hunk 1]
```diff
@@ -84,6 +85,36 @@ public class ContextMenuFlyoutItemViewModel
 		public bool ShowLoadingIndicator { get; set; }
 
 		public bool IsHidden { get; set; }
+
+		public ContextMenuFlyoutItemViewModel()
+		{
+		}
+		public ContextMenuFlyoutItemViewModel(IRichCommand command)
```
Comment (yaira2) [inline src/Files.App/ViewModels/ContextMenuFlyoutItemViewModel.cs:92]: Small nit, can you add an empty line?

**Comment by cinqmilleans** [comment]
I haven't looked at this class but probably it's possible. I will watch for the next iteration.

File: src/Files.App/UserControls/SidebarControl.xaml.cs [hunk 2]
```diff
@@ -208,16 +203,12 @@ private List<ContextMenuFlyoutItemViewModel> GetLocationItemMenuItems(INavigatio
 					Command = RestoreLibrariesCommand,
 					ShowItem = options.IsLibrariesHeader
 				},
-				new ContextMenuFlyoutItemViewModel()
+				new ContextMenuFlyoutItemViewModel(commands.EmptyRecycleBin)
 				{
-					Text = "BaseLayoutContextFlyoutEmptyRecycleBin/Text".GetLocalizedResource(),
-					Glyph = "\uEF88",
-					GlyphFontFamilyName = "RecycleBinIcons",
```
Comment (yaira2) [inline src/Files.App/UserControls/SidebarControl.xaml.cs:215]: Is the font family name still needed?

File: src/Files.App/Helpers/ContextFlyoutItemHelper.cs [hunk 3]
```diff
@@ -565,14 +563,10 @@ private static bool Check(ContextMenuFlyoutItemViewModel item, CurrentInstanceVi
...
-				new ContextMenuFlyoutItemViewModel()
+				new ContextMenuFlyoutItemViewModel(commands.EmptyRecycleBin)
 				{
-					Text = "BaseLayoutContextFlyoutEmptyRecycleBin/Text".GetLocalizedResource(),
-					Glyph = "\uEF88",
-					GlyphFontFamilyName = "RecycleBinIcons",
-					Command = commandsViewModel.EmptyRecycleBinCommand,
+					IsEnabled = RecycleBinHelpers.RecycleBinHasItems(),
 					ShowItem = !itemsSelected && currentInstanceViewModel.IsPageTypeRecycleBin,
-					ShowInRecycleBin = true,
```
Comment (yaira2) [inline src/Files.App/Helpers/ContextFlyoutItemHelper.cs:575]: Why was this removed?

**Comment by yaira2** [comment]
@cinqmilleans please fill out validation steps and add any other steps you used. This helps us know what to test for when reviewing the PR.

File: src/Files.App/Helpers/ContextFlyoutItemHelper.cs (hunk 3 above)
Comment (cinqmilleans) [reply src/Files.App/Helpers/ContextFlyoutItemHelper.cs:575]: ShowInThing properties are set to true in the constructor. The ShowItem property has the final say on whether the item is displayed or not. Here it is enough since ShowItem tests IsPageTypeRecycleBin. With contexts, ShowItem will no longer be needed because ShowItem is initialized with IsExecutable in the constructor.

File: src/Files.App/Actions/FileSystem/EmptyRecycleBinAction.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Files.App.Actions
+{
+	internal class EmptyRecycleBinAction : IAction
+	{
+		public string Label { get; } = "EmptyRecycleBin".GetLocalizedResource();
+
+		public RichGlyph Glyph { get; } = new RichGlyph("\uEF88", fontFamily: "RecycleBinIcons");
+
+		public async Task ExecuteAsync() => await RecycleBinHelpers.EmptyRecycleBin();
```
Comment (d2dyno1) [inline src/Files.App/Actions/FileSystem/EmptyRecycleBinAction.cs:14]: Could you add brackets `{ ... }`?

File: src/Files.App/Commands/RichGlyph.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	{
+		public static RichGlyph None { get; } = new(string.Empty);
+
+		public bool IsNone => string.IsNullOrEmpty(BaseGlyph);
+
+		public string BaseGlyph { get; }
+		public string OverlayGlyph { get; }
+		public string FontFamily { get; }
+
+		public RichGlyph(string baseGlyph, string overlayGlyph = "", string fontFamily = "")
```
Comment (d2dyno1) [inline src/Files.App/Commands/RichGlyph.cs:18]: Same here, and in other places too

File: src/Files.App/Actions/FileSystem/EmptyRecycleBinAction.cs (hunk 4 above)
Comment (cinqmilleans) [reply src/Files.App/Actions/FileSystem/EmptyRecycleBinAction.cs]: It's done.

File: src/Files.App/Commands/RichGlyph.cs (hunk 5 above)
Comment (cinqmilleans) [reply src/Files.App/Commands/RichGlyph.cs:18]: It's done.

**Review by yaira2 — APPROVED** [review]
LGTM

**Review by d2dyno1 — APPROVED** [review]
